desktop-src/dataxchg/html-clipboard-format.md

```diff
@@ -21,244 +21,346 @@ ms.date: 05/31/2018
+* `EndFragment`: Offset (in bytes) from the beginning of the clipboard to the end of the fragment.
+* `StartSelection`: Optional. Offset (in bytes) from the beginning of the clipboard to the start of the selection.
+* `EndSelection`: Optional. Offset (in bytes) from the beginning of the clipboard to the end of the selection.
 
-Other information of this kind could be added here later, since the HTML starts at the StartHTMLoffset. For example, multiple pairs of StartFragment / EndFragment could be added later to support noncontiguous selection of fragments.
+The `StartSelection` and `EndSelection` keywords are optional and must both be omitted if you do not want the application to generate this information.
 
-For the convenience of the programs generating the bytecounts, bytecounts could be left padded by zeros. For example, programs generating the bytecounts could arbitrarily affect ten (10) zeros to each keyword (StartHTML: 0000000000) and then, when the exact StartHTML bytecount is known (say, 71), the program can replace the appropriate number of zeros by the bytecount (StartHTML: 0000000071).
+Future revisions of the `CF_HTML` clipboard format may extend the header, for example, since the HTML starts at the `StartHTML` offset then multiple `StartFragment` and `EndFragment` pairs could be added later to support noncontiguous selection of fragments.
 
-The only character set supported by the clipboard is Unicode in its UTF-8 encoding. Because the first characters of UTF-8 and ASCII match, the description is always ASCII, but the bytes of the context (starting at StartHTML) could be using any other characters coded in UTF-8.
+### Offset syntax
 
-End of lines in the clipboard format header could be CR or CR/LF or LF.
+For the convenience of the programs generating the byte offsets the offset values may be optionally left-padded with an arbitrary amount of zero digits `'0'`, the reason for this is that programs sniffing the HTML for the offsets could write ten (10) zeros to its output buffer for each keyword (e.g. `StartHTML: 0000000000`) and later then when the exact `StartHTML` offset is known (say, 71), then program can overwrite the rightmost zeroes with "71" in the buffer (e.g. resulting in `StartHTML: 0000000071`).
 
-The fragment contains pure, valid HTML representing the area the user has selected (to Copy, for example). This contains the selected text plus the opening tags and attributes of any element that has an end tag within the selected text, and end tags at the end of the fragment for any start tag included. This is all information required for basic pasting of an HTML fragment.
+The only character set supported by the clipboard is Unicode in its UTF-8 encoding. Because the first characters of UTF-8 and ASCII match, the description is always ASCII, but the bytes of the context (starting at `StartHTML`) could be using any other characters coded in UTF-8.
```

Comment:
```suggestion
The only character set supported by the clipboard is Unicode (UTF-8). Because the first characters of UTF-8 and ASCII match, the description is always ASCII, but the bytes of the context (starting at `StartHTML`) could be using any other characters encoded in UTF-8.
```